### PR TITLE
refactor gpconfig file i/o to use python only

### PR DESCRIPTION
**Pull request by dkrieger-pvtl**
Previously, gpconfig used bash and Perl to manipulate the master and segment postgresql.conf files.  This made it hard to understand the transformation on string valued gucs.  We now just use python file i/o directly.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: gpMgmt/sbin/gpconfig_helper.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            # use a single write to make the file write atomic
+            if potential_match.lstrip().startswith(name):
+                outfile.write('#')
+            outfile.write(line)
+            new_lines = new_lines + 1
+        outfile.flush()
+
+    # atomically overwrite the existing file
+    if new_lines == len(lines):
+        os.rename(os.path.abspath(temp_conf_path), filename)
```
Comment (slari-pivotal) [inline gpMgmt/sbin/gpconfig_helper.py]: Two simultaneous updates of the postgresql.conf can still overwrite each other, but that is hard to protect against without a lock file.

**Comment by dkrieger-pvtl** [comment]
All our file writing is in `gpconfig_helper.py`.  If two users call `gpconfig` at the same time, we don't defend against that and could end up writing a file that we're reading...I think this is only true of `comment_paramter`...do we care?

File: gpMgmt/sbin/gpconfig_helper.py (hunk 1 above)
Comment (dkrieger-pvtl) [reply gpMgmt/sbin/gpconfig_helper.py]: That particular case is not exposed by this line...the `os.rename` is atomic so you'll get one of the two new files.  But there are a host of other race conditions....see below.

File: gpMgmt/bin/gpconfig [hunk 2]
```diff
@@ -253,21 +253,14 @@ def _show_all_segment_values_always(options):
...
-        else:
-            directory_string = seg.datadir
-
         print_verbosely(options, hostname, seg.hostname, seg.datadir)
-
-    cmd = GpAddConfigScript(hostname, directory_string, options.entry, value=value, removeonly=options.remove,
-                            ctxt=REMOTE,
-                            remoteHost=hostname)
-    pool.addCommand(cmd)
+        cmd_name = "add %s parameter on host %s for seg %s" % (options.entry, hostname, seg.datadir)
```
[the remark this answers is not in the document]
Comment (dkrieger-pvtl) [reply gpMgmt/bin/gpconfig]: Good idea; I put in a `FIXME` as this involves encoding issues in value that we'll handle next.

**Comment by dkrieger-pvtl** [comment]
And, yes, I'll push in the master pipeline with/without asserts as well....

**Pull request merged**